fs/fs_zenfs.cc

```diff
@@ -823,6 +840,128 @@ Status ZenFS::DecodeFileDeletionFrom(Slice* input) {
   return Status::OK();
 }
 
+/* find the latest valid snapshot record */
+Status ZenFS::RecoverFromSnapshotZone(ZenMetaLog* log) {
```

Comment:
Please merge these three functions: RecoverFromSnapshotZone, RecoverFromOpLogZone and RecoverFrom, at least they should use the same entry function.